kafka/signed_token_redeem_handler.go

```diff
@@ -111,88 +138,130 @@ func SignedTokenRedeemHandler(
+			message := fmt.Sprintf(
+				"Request %s: Failed to check redemption equivalence",
+				tokenRedeemRequestSet.Request_id,
+			)
 			return utils.ProcessingErrorFromErrorWithMessage(err, message, msg, logger)
 		}
+
 		// If the discovered equivalence is not one of the tolerableEquivalence
 		// options this redemption is considered a duplicate.
 		if !containsEquivalnce(tolerableEquivalence, equivalence) {
-			logger.Error().Msg(fmt.Sprintf("Request %s: Duplicate redemption: %e", tokenRedeemRequestSet.Request_id, err))
+			logger.Error().Msg(fmt.Sprintf(
+				"Request %s: Duplicate redemption: %e",
+				tokenRedeemRequestSet.Request_id,
+				err,
+			))
 			redeemedTokenResults = append(redeemedTokenResults, avroSchema.RedeemResult{
 				Issuer_name:     "",
 				Issuer_cohort:   0,
-				Status:          DUPLICATE_REDEMPTION,
+				Status:          avroSchema.RedeemResultStatusDuplicate_redemption,
 				Associated_data: request.Associated_data,
 			})
 			continue
 		}
+
 		if err := server.PersistRedemption(*redemption); err != nil {
-			logger.Error().Err(err).Msg(fmt.Sprintf("Request %s: Token redemption failed", tokenRedeemRequestSet.Request_id))
+			logger.Error().Err(err).Msg(fmt.Sprintf(
+				"Request %s: Token redemption failed",
+				tokenRedeemRequestSet.Request_id,
+			))
 			redeemedTokenResults = append(redeemedTokenResults, avroSchema.RedeemResult{
 				Issuer_name:     "",
 				Issuer_cohort:   0,
-				Status:          ERROR,
+				Status:          avroSchema.RedeemResultStatusError,
 				Associated_data: request.Associated_data,
 			})
 			continue
 		}
-		logger.Info().Msg(fmt.Sprintf("Request %s: Redeemed", tokenRedeemRequestSet.Request_id))
+
+		logger.Info().Msg(fmt.Sprintf(
+			"Request %s: Redeemed",
+			tokenRedeemRequestSet.Request_id,
+		))
 		issuerName := verifiedIssuer.IssuerType
 		redeemedTokenResults = append(redeemedTokenResults, avroSchema.RedeemResult{
 			Issuer_name:     issuerName,
 			Issuer_cohort:   verifiedCohort,
-			Status:          OK,
+			Status:          avroSchema.RedeemResultStatusOk,
 			Associated_data: request.Associated_data,
 		})
 	}
+
 	resultSet := avroSchema.RedeemResultSet{
 		Request_id: tokenRedeemRequestSet.Request_id,
 		Data:       redeemedTokenResults,
 	}
 	var resultSetBuffer bytes.Buffer
 	err = resultSet.Serialize(&resultSetBuffer)
 	if err != nil {
-		message := fmt.Sprintf("Request %s: Failed to serialize ResultSet", tokenRedeemRequestSet.Request_id)
+		message := fmt.Sprintf(
+			"Request %s: Failed to serialize ResultSet",
+			tokenRedeemRequestSet.Request_id,
+		)
 		return utils.ProcessingErrorFromErrorWithMessage(err, message, msg, logger)
 	}
 
 	err = Emit(producer, resultSetBuffer.Bytes(), logger)
 	if err != nil {
-		message := fmt.Sprintf("Request %s: Failed to emit results to topic %s", tokenRedeemRequestSet.Request_id, producer.Topic)
+		message := fmt.Sprintf(
```

Comment:
we should increase Prometheus counters on these unexpected cases and wire alert manager to notify us.

Reply:
right before all `return utils.ProcessingErrorFromErrorWithMessage(err, message, msg, logger)` instructions

Reply:
right before all `return utils.ProcessingErrorFromErrorWithMessage(err, message, msg, logger)` instructions